src/ripple/ledger/detail/RawStateTable.h

```diff
@@ -0,0 +1,91 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_LEDGER_RAWSTATETABLE_H_INCLUDED
+#define RIPPLE_LEDGER_RAWSTATETABLE_H_INCLUDED
+
+#include <ripple/ledger/RawView.h>
+#include <ripple/ledger/ReadView.h>
+#include <map>
+#include <utility>
+
+namespace ripple {
+namespace detail {
+
+// Helper class that buffers raw modifications
```

Comment:
Its a good question and one that certainly should be asked. I tried to tweak the implementation so it could be used for both. But while there was less total code, it was harder to audit. So this is a judgement call. The operation of this logic is critical enough that we want to simply make the classes separate so they can do the things they need to do differently.